### DISCUSSION TITLE
Glossary and tooltips

### DISCUSSION THREAD

**Discussion by nichtich** [quarto-dev/quarto-cli#805] [Feature Requests]
Creating a book with quarto I'd like to add a glossary of acronyms and have terms from the glossary exlained by tooltips or margin notes. Any idea how to best do this? Some ideas:

* put glossary in a CSV file
* includes the glossary as markdown definition list by
  * creating a lua shortcode that reads the CSV file
  * or creating a filter that converts the CSV to markdown definition list
* either create a shortcode to markup use of an acronym `{{< acronym XYZ >}}`
* or mark up like this `` `XYZ`{.acronym} `` and convert with a filter

or

* write glossary in markdown and use shortcodes and/or markup as above

But how/where to implement the tooltips for HTML output? What do you think?

**Comment by jjallaire** [comment]
I think you will want to write a Lua filter that runs before the quarto filters and either encloses definitions in `column-margin` divs or adds css classes to the terms that you then use JavaScript to implement the tooltips with (note that both tippy.js and popper.js are already included with quarto so you can use their APIs to create the popovers.

**Comment by nichtich** [comment]
I've come up with the following solution:

* Manage [glossary in a YAML file](https://github.com/pro4bib/handbuch-it-in-bibliotheken/blob/main/glossar.yml) that is included as additional metadata file
* Add [filter that reads the glossary entries](https://github.com/pro4bib/handbuch-it-in-bibliotheken/blob/main/templates/glossary.lua), inserts full glossary at `{{< glossary >}}` and replaces emphasized terms with `<abbr>` if found in the glossary

Note the filter could not be implemented as shortcode, as shortcodes cannot access full nested metadata (only flattened variant), so it's a normal Pandoc lua filter.

**Reply by debruine** [reply]
See https://debruine.github.io/glossary/ for an R package I've written to implement a glossary with a YAML file. (I still need to figure out LUA filters for a better implementation that isn't reliant on R)

**Comment by mcanouil** [comment]
Closing in favour of https://github.com/quarto-dev/quarto-cli/issues/1697 (for the tooltip part, better to comment on the issue to centralise request on the same feature)
